### DISCUSSION TITLE
Hololens 2: How to configure and ping a static IP for ethernet connection?

### DISCUSSION THREAD

**Discussion by A-Ivan** [microsoft/MixedRealityToolkit-Unity#10415] [Q&A] [answered]
I'm having issues trying to set a static IP for the ethernet connection on a Hololens 2 (build 20348.1442).

1. I went through the [settings app](https://docs.microsoft.com/en-us/hololens/hololens-new-settings) > Network & Internet > Ethernet > Network
2. IP Settings > Edit
3. Change Automatic (DHCP) to Manual
4. Toggle IPv4 to On
5. Input desired IP static address, Subnet mask, Gateway, Preferred DNS
6. Click Save

To make sure the IP was configured correctly, I checked the Device Portal information on Hololens (Settings > Update & Security > For Developers > Device Portal) and the Ethernet 2 connection is shown with the static IP I configured.

Using an ethernet to USB-C connector to connect a computer to Hololens, I tried to ping the Hololens ethernet IP (the one I just configured), but no packets are being received.

I'm not sure if something was not configured correctly or if there are any issues with the firewall in Hololens.

I tried changing the Hololens ethernet network from public to private, but this does not solve the issue. There is a link beneath the private option "Configure firewall and security settings", clicking this opens the Microsoft Store and seems to try to install Windows Defender, but this does not work.

Is there a way to run command prompt or some other form of terminal in Hololens? I want to try to ping the computer from Hololens to see if at least that connection is working.

**Comment by A-Ivan** [comment, marked as answer]
I got it to work, not sure what exactly solved the issue though.

What I did: I restarted Hololens and I set the Preferred DNS encryption (in the IPv4 setup window) to: "Unencrypted only".

I wasn't able to ping Hololens, but my application which had ethernet network connection is now working correctly.

You can use any DNS address and it doesn't matter if the connection is set to public or private, it works on both.

I hope this helps someone else at least.

p.s. Going through the menus on Hololens to setup static IPs for Wi-Fi and Ethernet connections is a bit of a pain. Would be great to be able configure this through the Windows Device Portal on a computer.

**Comment by polar-kev** [comment]
Thanks for posting your solution @A-Ivan. It should help others who run into this same problem.

**Reply by A-Ivan** [reply]
You're welcome @polar-kev .

Just to be clear, there is no way to configure Hololen's IP address settings through a Windows computer (i.e., using the Windows Device Portal), correct?

There is also no way to ping the Hololens from a command prompt in Windows or vice-versa?
